cmd/seed_server/app/firewall.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/gardener/vpn2/pkg/utils"
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+)
+
+func firewallCommand() *cobra.Command {
+	var (
+		device string
+		mode   string
+	)
+
+	cmd := &cobra.Command{
+		Use:   "firewall",
+		Short: "firewall",
+		Args:  cobra.ArbitraryArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			log, err := utils.InitRun(cmd, Name+"-firewall")
+			if err != nil {
+				return err
+			}
+			return runFirewallCommand(log, device, mode)
+		},
+	}
+
+	cmd.Flags().StringVar(&device, "device", "", "device to configure")
+	cmd.Flags().StringVar(&mode, "mode", "", "mode of firewall (up or down)")
+	cmd.MarkFlagsRequiredTogether("device", "mode")
+
+	return cmd
+}
+
+func runFirewallCommand(log logr.Logger, device, mode string) error {
+	// Firewall subcommand is called indirectly from openvpn. As PATH env variables seems not to be set,
+	// it is injected here.
+	os.Setenv("PATH", "/sbin")
+	iptable4, err := iptables.New(iptables.IPFamily(iptables.ProtocolIPv4))
```

Comment:
We added an iptables backend detection with https://github.com/gardener/vpn2/pull/63.
We have seen issues on some os versions, that not all required modules are present in the vpn-seed-server container.
How is this handled in `github.com/coreos/go-iptables/iptables`? Is there a similar check to determine the iptables backend?

Reply:
Looking into the code of `github.com/coreos/go-iptables/iptables`, it always uses either `iptables` or `ip6tables` command. It has a detection for nft or legacy mode, but it is not used to set the iptables executable name.

@axel7born
Can you explain how to reproduce the issue you fixed with #63 ?

Reply:
We have seen the issue with gardenlinux 1312.0.0.  There the default switched from iptables-legacy to iptables-nft.
The vpn container still used iptables-legacy and for the ipv6 case not all required kernel modules were present. I can't reproduce the isssue with newer gardenlinux versions. 
We have also seen the issue in gke soil clusters. But also there, I can't reproduce it now.
To reproduce the issue, exec into a vpn pod, and try to run the commands `ip6tables-legacy -L` and `ip6tables-nft -L`. When the needed kernel modules are not present, the command returns an error.
When we just run `iptables` there is a risk that the command doesn't work due to missing kernel modules.

Reply:
Added iptables backend detection, see below